pkg/model/iam/iam_builder.go

```diff
@@ -1118,6 +1123,23 @@ func addAmazonVPCCNIPermissions(p *Policy, resource stringorslice.StringOrSlice,
 	)
 }
 
+func addNodeTerminationHandlerSQSPermissions(p *Policy, resource stringorslice.StringOrSlice) {
```

Comment:
Given that this is assigned to nodes, it may be a bit broad.

One option is to put the deployment on master nodes instead and add the permissions there. some of the below should already be in place.
Another is to let this depend on the coming IRSA support. That could look like something I did here: https://github.com/kubernetes/kops/pull/11088

SQS permissions should maybe be limited to only to the NTH resources.

Reply:
IRSA seems like the better way to go to me